File: src/jvmTest/java/org/snakeyaml/engine/v2/tokens/TokenTest.java
/*
 * Copyright (c) 2018, SnakeYAML
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */
package org.snakeyaml.engine.v2.tokens;

import it.krzeminski.snakeyaml.engine.kmp.tokens.ScalarToken;
import it.krzeminski.snakeyaml.engine.kmp.tokens.Token;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

@org.junit.jupiter.api.Tag("fast")
class TokenTest {

  @Test
  void testToString() {
    Token token = new ScalarToken("a", true, null, null);
    assertEquals("<scalar> plain=true style=: value=a", token.toString());
  }
}

File: src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/tokens/TokenTest.kt
package it.krzeminski.snakeyaml.engine.kmp.tokens

import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class TokenTest : FunSpec({

    test("toString should format scalar token correctly") {
        val token: Token = ScalarToken("a", true, null, null)
        token.toString() shouldBe "<scalar> plain=true style=: value=a"
    }
})
